test/tests/functional/pbs_hook_execjob_end.py

```diff
@@ -0,0 +1,261 @@
+        date_time1 = str1.split(";")[0]
+        epoch1 = int(time.mktime(time.strptime(
+            date_time1, '%m/%d/%Y %H:%M:%S')))
+        # following message should be logged while execjob_end hook is in sleep
+        (_, str1) = self.mom.log_match("executed exechost_periodic hook",
+                                       starttime=epoch1-1, endtime=epoch1+10,
+                                       n=100, max_attempts=10, interval=1)
+        date_time2 = str1.split(";")[0]
+        epoch2 = int(time.mktime(time.strptime(
+            date_time2, '%m/%d/%Y %H:%M:%S')))
+        (_, str1) = self.mom.log_match(
+            "Job;%s;execjob_end hook ended" %
+            jid, starttime=epoch2-1, n=100,
+            max_attempts=10, interval=2)
+        date_time3 = str1.split(";")[0]
+        self.logger.info(
+            "execjob_end hook executed at: %s,"
+            "exechost_periodic at: %s and execjob_end hook ended at: %s" %
+            (date_time1, date_time2, date_time3))
+
+    def test_execjob_end_hook_order_and_reject(self):
+        """
+        Test with multiple execjob_end hooks having different order
+        with one of the hooks rejecting the job.
+        """
+        hook_name1 = "execjob_end_logmsg1"
+        hook_body = ("import pbs\n"
+                     "e = pbs.event()\n"
+                     "pbs.logjobmsg(e.job.id, \
+                                  'executed %s hook' % e.hook_name)\n"
+                     "e.accept()\n")
+        attr = {'event': 'execjob_end', 'order': '1', 'enabled': 'True'}
+        self.server.create_import_hook(hook_name1, attr, hook_body)
+        hook_name = "execjob_end_logmsg2"
+        hook_body1 = ("import pbs\n"
+                      "e = pbs.event()\n"
+                      "pbs.logjobmsg(e.job.id, 'executed execjob_end hook')\n"
+                      "e.reject('Job is rejected')\n")
+        attr = {'event': 'execjob_end', 'order': '2', 'enabled': 'True'}
+        self.server.create_import_hook(hook_name, attr, hook_body1)
+        hook_name2 = "execjob_end_logmsg3"
+        attr = {'event': 'execjob_end', 'order': '170', 'enabled': 'True'}
+        self.server.create_import_hook(hook_name2, attr, hook_body)
+        j = Job(TEST_USER)
+        j.set_sleep_time(1)
+        jid = self.server.submit(j)
+        self.mom.log_match("Job;%s;executed %s hook" % (jid, hook_name1),
+                           n=100, max_attempts=10, interval=2)
+        self.mom.log_match("Job;%s;Job is rejected" % jid,
+                           n=100, max_attempts=10, interval=2)
+        self.mom.log_match("Job;%s;executed %s hook" % (jid, hook_name2),
+                           n=100, max_attempts=10, interval=2, existence=False)
+
+    def test_execjob_end_multi_job(self):
+        """
+        Test to make sure that mom is unblocked with
+        execjob_end hook with mutiple jobs
+        """
+        pltfom = self.du.get_platform()
+        if pltfom != 'cray':
```

Comment:
not just cray but this won't work on a cpuset/cgroups enabled system as well. how about you just check if cpus is greater than 2 then run the test otherwise skip.

Reply:
Thanks! Addressed it in the new commit.